st2-pastebin/sensors/pastebin_poller.py

```diff
@@ -0,0 +1,135 @@
+    if ipversion == 6:
+        socket.getaddrinfo = getaddrinfoIPv6
+    else:
+        socket.getaddrinfo = getaddrinfoIPv4
+    return requests.get(url)
+
+
+class PasteBinPoller(PollingSensor):
+    """ regularly polls the pastebin scrape API endpoint and reports back new keys """
+
+    def __init__(self, sensor_service, config=None, poll_interval=None):
+        """ sets up the thing """
+        super(PasteBinPoller, self).__init__(sensor_service=sensor_service,
+                                             config=config, poll_interval=poll_interval)
+        self._trigger_ref = 'pastebin.new_paste'
+        self._logger = self._sensor_service.get_logger(__name__)
+        self._last_time = None
+        self._url = "{}?limit={}".format(SCRAPE_URL, self._config['poll_maxresults'])
+
+    def setup(self):
+        """ Setup stuff goes here. This is called only once by the system."""
+        self.set_poll_interval(self._config['poll_interval'])
+        self._logger.debug('set poll interval to {}'.format(self._config['poll_interval']))
+
+    def poll(self):
+        """ does the polling action
+            if it finds a "new" paste, emits a payload:
+                {
+                    'date' : int,
+                    'key' : str,
+                    'title' : str,
+                    'user' : str,
+                    'size' : int,
+                    'syntax' : str,
+                }
+        """
+        self.set_poll_interval(self._config['poll_interval'])
+        self._logger.debug('set poll interval to {}'.format(self._config['poll_interval']))
+        
+        
+        try:
+            # do the HTTP request
+            self._logger.debug("Doing the request to {}".format(self._url))
+            req = request_get_versioned(self._url, self._config['ipversion'])
+
+            if req and req.status_code == 200:
+                self._logger.debug("Got a response from the API")
+                try:
+                    jsondata = req.json()
+                except TypeError:
+                    self._logger.debug("JSON Decode failed, stopping. Probably not running from a whitelisted IP")
+                    return
+                if "VISIT: https://pastebin.com/doc_scraping_api TO GET ACCESS!" in req.text:
+                    self._logger.debug("Our source IP is not whitelisted, stoppping.")
+                    return
+                # sort by timestamp, it comes in most-recent-first
+                data = sorted(jsondata, key=lambda k: k['date'])
+                for paste in data:
+                    self._logger.debug("time:{} key:{}".format(paste['date'], paste['key']))
+                    if paste['date'] > self._get_last_time():
+                        # this is the timestamp of the last processed paste
+                        self._set_last_time(last_time=paste['date'])
+                        # do the thing
+                        payload = {'date' : int(paste['date']), 
```

Comment:
any reason you don't just include all of the fields from the response (curious)?

Reply:
Because I needed to convert the formats.